ratis-server/src/main/java/org/apache/ratis/server/raftlog/RaftLog.java

```diff
@@ -84,19 +85,24 @@
   private final Runner runner = new Runner(this::getName);
   private final OpenCloseState state;
   private final RaftLogMetrics raftLogMetrics;
+  private final LongSupplier snapshotIndexSupplier;
```

Comment:
Since we already have snapshotIndex, let's rename this to
```
  private final LongSupplier getSnapshotIndexFromStateMachine;
```

Reply:
Sure, will do.